src/app/components/Notifications/NotificationTransactionItem.tsx

```diff
@@ -17,21 +12,22 @@ export const NotificationTransactionItem = ({
 	containmentRef,
 	onTransactionClick,
 }: NotificationTransactionItemProps) => {
-	const [transaction, setTransaction] = useState<TransactionData>();
-	const [wallet, setWallet] = useState<ReadWriteWallet>();
+	const [transaction, setTransaction] = useState<ExtendedTransactionData>();
+	const [walletName, setWalletName] = useState<string>();
 
 	useEffect(() => {
 		const fetchTransaction = async () => {
 			try {
 				/* @ts-ignore */
-				const wallet = profile.wallets().findByAddress(notification.meta?.address);
-				if (!wallet) return;
+				// TODO: Fetch transactions by their ids
+				const receivedTxs = await profile.transactionAggregate().receivedTransactions({ cursor: 1, limit: 15 });
```

Comment:
You can't manually pass in a cursor because every blockchain uses different cursors. Some use numbers, UUIDs, Timestamps and so on. You will have to either ignore the cursor or pass in the cursor for the next/previous set of data from the previous request.

Reply:
@faustbrian exactly yes, this is error-prone. I temporarily use it to finish up the functionality with fixture data. At this example, using transaction id(s) as query parameters would be convenient from sdk (in profile scope), because we already know which transactions we want to fetch.

Reply:
Data that is needed for notifications should be stored with the notifications themselves instead of sending network requests to get that data. Notifications are intended to be static data.

Reply:
Yeah, true, data should and is static for regular notifications (e.g plugin updates). The case we are facing is with transaction-notifications only where the user clicks it and transaction detail modal is opened. We need to show updated data in that case (e.g confirmation status). Also I could have used `wallet.client.transaction` but our transaction detail modals require ExtendedTransactionData to work properly  :hmhm